### PR TITLE
Pin 2095

### PR DESCRIPTION
**Pull request by TonioGela**
This PR depends on both https://github.com/pagopa/interop-be-tenant-management/pull/43 and https://github.com/pagopa/interop-be-tenant-process/pull/31 

Here were added:
- the `/tenants/{tenantId}` endpoint in POST to update the tenant's mail (exposed as a single one and not like in the persistence model where is a list of mails)
- the `/eservices/{eserviceId}/descriptor/{descriptorId}` endpoint in GET that retrieves the information from the catalog process and decorates it using the tenant-management informations.

### CODE REVIEW

File: src/main/resources/interface-specification.yml [hunk 1]
```diff
@@ -2490,9 +2490,178 @@ paths:
...
+            'X-Rate-Limit-Remaining':
+              schema:
+                type: integer
+              description: Remaining requests within time interval
+            'X-Rate-Limit-Interval':
+              schema:
+                type: integer
+              description: Time interval in milliseconds. Allowed requests will be constantly replenished during the interval. At the end of the interval the max allowed requests will be available
+        '404':
+          description: Tenant not found
```
Comment (galales) [inline src/main/resources/interface-specification.yml]:
```suggestion
          description: EService not found
```

**Review by galales — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EservicesApiServiceImpl.scala [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  override def getEservice(eserviceId: String)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerEService: ToEntityMarshaller[EService]
+  ): Route = {
+    val result: Future[EService] = for {
+      eServiceUUID    <- eserviceId.toFutureUUID
+      catalogEservice <- catalogProcessService.getEServiceById(eServiceUUID)
+      (origin, code) = (catalogEservice.producer.id, catalogEservice.producer.name)
```
Comment (beetlecrunch) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EservicesApiServiceImpl.scala]: This extraction is wrong: `catalogEservice.producer.id` is not the **origin** and `catalogEservice.producer.name` is not the **code**.
This information comes from the tenant so, if I don't miss something, you should call tenantManagementService using the internal id that is the `catalogEservice.producer.id`.

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EservicesApiServiceImpl.scala [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    Logger.takingImplicit[ContextFieldsToLog](this.getClass)
+
+  override def getEservice(eserviceId: String)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerEService: ToEntityMarshaller[EService]
+  ): Route = {
+    val result: Future[EService] = for {
+      eServiceUUID    <- eserviceId.toFutureUUID
+      catalogEservice <- catalogProcessService.getEServiceById(eServiceUUID)
```
Comment (beetlecrunch) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EservicesApiServiceImpl.scala]:
```suggestion
      eService <- catalogProcessService.getEServiceById(eServiceUUID)
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EservicesApiServiceImpl.scala (hunk 2 above)
Comment (TonioGela) [reply src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EservicesApiServiceImpl.scala]: Thank you, I had in the plans to ask you about the correctness of this extraction

**Review by beetlecrunch — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

File: src/main/resources/interface-specification.yml [hunk 4]
```diff
@@ -2519,6 +2702,91 @@ components:
       schema:
         type: string
   schemas:
+    Mail:
+      type: object
+      required:
+        - address
```
Comment (galales) [inline src/main/resources/interface-specification.yml:2801]: I know you know, but 
```suggestion
      required:
        - address
        - description
```

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EservicesApiServiceImpl.scala [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    toEntityMarshallerEServiceDescriptor: ToEntityMarshaller[EServiceDescriptor],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route = {
+    val result: Future[EServiceDescriptor] = for {
+      eServiceUUID   <- eserviceId.toFutureUUID
+      descriptorUUID <- descriptorId.toFutureUUID
+      eService       <- catalogProcessService.getEServiceById(eServiceUUID)
+      descriptor     <- eService.descriptors
+        .find(_.id == descriptorUUID)
+        .toFuture(new Exception("not found"))
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EservicesApiServiceImpl.scala]: ⚠️

File: src/main/resources/interface-specification.yml (hunk 4 above)
Comment (TonioGela) [reply src/main/resources/interface-specification.yml:2801]: The requirements were to have description optional

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EservicesApiServiceImpl.scala (hunk 5 above)
Comment (TonioGela) [reply src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EservicesApiServiceImpl.scala]: ☝️

**Review by galales — APPROVED** [review]

**Pull request merged**